src/material/legacy-dialog/testing/dialog-opener.ts

```diff
@@ -27,6 +27,11 @@ import {_MatTestDialogOpenerBase} from '@angular/material/dialog/testing';
   template: '',
   changeDetection: ChangeDetectionStrategy.OnPush,
   encapsulation: ViewEncapsulation.None,
+  host: {
+    // This binding is used to ensure that the component ID
+    // doesn't clash with the `MatTestDialogOpener`.
```

Comment:
do you think we could have special comments in here that the script could use to tell us if e.g. this trick suddenly is no longer needed? e.g. when dialog test opener changes

Reply:
I think that we could, but I would probably lean more towards a config file. I'm not sure if it's worth the trouble now though given that these workarounds are only there for the legacy components and their perf impact is minimal.

Reply:
yeah, fine by me if it's just legacy components 👍

Reply:
(although I can see menu here)